.github/workflows/Deploy_web_site_with_ssh.yml

```diff
@@ -0,0 +1,34 @@
+name: Deploy_web_site_with_ssh
+
+# read-only repo token
+# no access to secrets
+on:
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Change BASE_PATH
+        run: yq -i '.JB.BASE_PATH = "https://cgal.geometryfactory.com/~${{ secrets.SSH_USER }}/cgal-web"' _config.yml
+      - name: Remove demo/ (temporarily)
+        run: rm -rf demo/
+      - name: Build with Jekyll
+        uses: actions/jekyll-build-pages@v1
+        with:
+          source: ./
+          destination: ./_site
+      - name: install ssh keys
+        run: |
+          install -m 600 -D /dev/null ~/.ssh/id_rsa
+          echo "${{ secrets.SSH_PRIVATE_KEY }}" > ~/.ssh/id_rsa
+          ssh-keyscan -H ${{ secrets.SSH_HOST }} > ~/.ssh/known_hosts
+      - name: deploy website to ssh using rsync
+        run: rsync -avz ./_site/ ${{ secrets.SSH_USER }}@${{ secrets.SSH_HOST }}:/home/${{ secrets.SSH_USER }}/public_html/cgal-web
```

Comment:
I think only `secrets.SSH_PRIVATE_KEY` should be a secret. The other ones should be variables.

And the path `/home/${{ secrets.SSH_USER }}/public_html/cgal-web` should be configurable with a variable as well.